pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -412,7 +412,10 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1.PipelineRun, getPipel
 		pr.Status.MarkRunning(v1.PipelineRunReasonResolvingPipelineRef.String(), message)
 		return nil
 	case errors.Is(err, apiserver.ErrReferencedObjectValidationFailed), errors.Is(err, apiserver.ErrCouldntValidateObjectPermanent):
-		pr.Status.MarkFailed(v1.PipelineRunReasonFailedValidation.String(), err.Error())
+		logger.Errorf("Failed dryRunValidation for PipelineRun %s: %w", pr.Name, err)
```

Comment:
the error is logged when the reconciler returns this error, do we need to log it one more time here?

Reply:
I think we do because this adds more context for the failure. Prior to this, we would not know explicitly it failed dry run validation with ErrReferencedObjectValidationFailed or ErrCouldntValidateObjectPermanent.

Reply:
We can wrap the error with ""Failed dryRunValidation for PipelineRun %s: %w", pr.Name, err" then return it, the up level logger will log this error with the information we added?

Reply:
The apiServer here https://github.com/tektoncd/pipeline/blob/a7ab091dd766c6a26fbd592f46c0a6e4a82cd165/pkg/reconciler/apiserver/apiserver.go#L87 would not have the PipelineRun info.

And also this was implemented as is for [markFailed](https://github.com/tektoncd/pipeline/blob/a7ab091dd766c6a26fbd592f46c0a6e4a82cd165/pkg/apis/pipeline/v1/pipelinerun_types.go#L467) has the messageFormat as an input for the user-facing condition message. This is what we've previously done to make consistent with the log from my understanding. WDYT?